src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/PrisonerIepLevelReviewService.kt

```diff
@@ -50,24 +52,27 @@ class PrisonerIepLevelReviewService(
 
   suspend fun getPrisonerIepLevelHistory(
     bookingId: Long,
-    useNomisData: Boolean = true,
     withDetails: Boolean = true,
     useClientCredentials: Boolean = false,
   ): IepSummary {
-    return if (useNomisData) {
+    return if (!featureFlagsService.isIncentivesDataSourceOfTruth()) {
       prisonApiService.getIEPSummaryForPrisoner(bookingId, withDetails, useClientCredentials)
     } else {
-      buildIepSummary(prisonerIepLevelRepository.findAllByBookingIdOrderByReviewTimeDesc(bookingId), getIncentiveLevels(), withDetails)
+      val reviews = prisonerIepLevelRepository.findAllByBookingIdOrderByReviewTimeDesc(bookingId)
+      if (reviews.count() == 0) throw IncentiveReviewNotFoundException("No Incentive Reviews for booking ID $bookingId")
```

Comment:
Nice, I like the throwing of an exception when we detect a problematic state 👍 

Can I suggest we also log a debug message on top of this (can help debug these)

Reply:
Actually the exception is probably enough as we see that in AppInsights anyway so don't worry :)